pallets/oracle/src/oracle_mock.rs

```diff
@@ -0,0 +1,87 @@
+#[cfg(feature = "testing-utils")]
+use std::{cell::RefCell, sync::RwLock};
+
+use dia_oracle::{CoinInfo, DiaOracle};
+use orml_oracle::{DataProvider, TimestampedValue};
+use sp_runtime::traits::Convert;
+
+use crate::dia::MockDiaOracleConvertor;
+use sp_arithmetic::{FixedI128, FixedU128};
+pub type UnsignedFixedPoint = FixedU128;
+use primitives::oracle::Key;
+type Moment = u64;
+
+#[derive(Clone)]
+struct Data {
+	pub key: (Vec<u8>, Vec<u8>),
+	pub price: u128,
+	pub timestamp: u64,
+}
+
+thread_local! {
+	static COINS: RefCell<Vec<Data>> = RefCell::new(vec![]);
```

Comment:
This is a bit hacky. But if it works, fine for me since it is only used for testing purposes.